source/slang/glsl.meta.slang

```diff
@@ -407,13 +422,707 @@ ${{{{
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T, N> mod(vector<T, N> x, vector<T, N> y)
+{
+    return fmod(x, y);
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
+public T mix(T x, T y, T a)
+{
+    return lerp(x, y, a);
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T, N> mix(vector<T, N> x, vector<T, N> y, T a)
+{
+    return lerp(x, y, vector<T, N>(a));
+}
+
+__generic<T : __BuiltinFloatingPointType, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T, N> mix(vector<T, N> x, vector<T, N> y, vector<T, N> a)
+{
+    return lerp(x, y, a);
+}
+
+__generic<T>
+__target_intrinsic(glsl)
+[__readNone]
+public T mix(T x, T y, bool a)
+{
+    return (a ? y : x);
+}
+
+__generic<T, let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<T, N> mix(vector<T, N> x, vector<T, N> y, vector<bool, N> a)
+{
+    vector<T, N> result;
+    for (int i = 0; i < N; i++)
+    {
+        result[i] = (a[i] ? y[i] : x[i]);
+    }
+    return result;
+}
+
+__target_intrinsic(glsl)
+[__readNone]
+public int floatBitsToInt(highp float x)
+{
+    return asint(x);
+}
+
+__generic<let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<int, N> floatBitsToInt(highp vector<float, N> x)
+{
+    return asint(x);
+}
+
+__target_intrinsic(glsl)
+[__readNone]
+public uint floatBitsToUint(highp float x)
+{
+    return asuint(x);
+}
+
+__generic<let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<uint, N> floatBitsToUint(highp vector<float, N> x)
+{
+    return asuint(x);
+}
+
+__target_intrinsic(glsl)
+[__readNone]
+public float intBitsToFloat(highp int x)
+{
+    return asfloat(x);
+}
+
+__generic<let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<float, N> intBitsToFloat(highp vector<int, N> x)
+{
+    return asfloat(x);
+}
+
+__target_intrinsic(glsl)
+[__readNone]
+public float uintBitsToFloat(highp uint x)
+{
+    return asfloat(x);
+}
+
+__generic<let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<float, N> uintBitsToFloat(highp vector<uint, N> x)
+{
+    return asfloat(x);
+}
+
+__target_intrinsic(glsl)
+[__readNone]
+public float fma(float a, float b, float c)
+{
+    return mad(a, b, c);
+}
+
+__generic<let N : int>
+__target_intrinsic(glsl)
+[__readNone]
+public vector<float, N> fma(vector<float, N> a, vector<float, N> b, vector<float, N> c)
+{
+    return mad(a, b, c);
+}
+
+//
+// Section 8.5. Geometric Functions
+//
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
+[ForceInline]
+public T dot(T x, T y)
+{
+    return x * y;
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
+[ForceInline]
+public T faceforward(T n, T i, T ng)
+{
+    return dot(ng, i) < T(0.0f) ? n : -n;
+}
+
+__generic<T : __BuiltinFloatingPointType>
+__target_intrinsic(glsl)
+[__readNone]
```

Comment:
Same here. In general if HLSL.meta.slang is missing such an overload, let's just add the function there for completeness.